Pirat.DatabaseTests/ResourceChangeTest.cs

```diff
@@ -80,68 +80,102 @@ public void Dispose()
             Assert.Null(exception);
         }
 
-        [Fact(Skip = "TODO")]
-        public async void Test_ChangeProviderInformation_Possible()
+        /// <summary>
+        /// Tests if provider information can be changed in the database
+        /// </summary>
+        [Fact]
+        public async Task Test_ChangeProviderInformation_Possible()
         {
+            //Create a provider and change attributes
             Provider provider = _captainHookGenerator.GenerateProvider();
             provider.name = "Peter Pan";
             provider.phone = "987766";
             provider.organisation = "Never Grow Up Kids";
             provider.address.postalcode = "88888";
             provider.address.country = "Atlantis";
 
-            Exception exception = await Record.ExceptionAsync(() => _resourceUpdateService.ChangeInformation(_token, provider));
-            Assert.Null(exception);
+            //Update
+            var changedRows = await _resourceUpdateService.ChangeInformation(_token, provider);
+            Assert.True(changedRows == 2);
 
+            //We take a device to make a query and from this query we also get the changed provider
             Device device = _captainHookGenerator.GenerateDevice();
             var response = await _resourceDemandService.QueryOffers(device);
+
+            //Assert
             Assert.NotNull(response);
             Assert.NotEmpty(response);
             Provider providerFromDevice = response.First().provider;
             Console.Out.WriteLine(providerFromDevice);
             Console.Out.WriteLine(provider);
-            Assert.True(providerFromDevice.Equals(provider));
+            //TODO: 1. Provider is null because so far everything is non public
+            //TODO: 2. If the provider is non-public we should create a provider object with empty attributes instead
```

Comment:
Wait, why that? Please don't start sending some empty provider objects to the frontend. If the provider is non-public, null would be - in my opinion - the correct value.